### PR TITLE
Idva5 1348 data to be stored on data wrapper

### PR DESCRIPTION
**Pull request by atedros-ch**
- Mongo DB data wrapper added
- unit tests updated

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/acsp/service/AcspService.java [hunk 1]
```diff
@@ -162,7 +175,8 @@ public ResponseEntity<Object> getAcspApplicationStatus(String userId, String req
                 LOGGER.info("No application found for userId: " + userId);
                 return new ResponseEntity<>(HttpStatus.NOT_FOUND);
             }
-            String transactionId = application.get().getLinks().get("self").split("/")[2];
+
+            String transactionId = application.get().getData().getLinks().get("self").split("/")[2];
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/service/AcspService.java:179]: I know this is old line... but transactionId can be obtained from transaction object. We dont need to do these many dot operations

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by sampankumar — APPROVED** [review]

**Review by swalke-ch — APPROVED** [review]

**Pull request merged**